### PR TITLE
Signalification effort

### PR DESCRIPTION
**Pull request by mjtorn**
This is a loose bag of commits I made while refactoring the code for signal support.

Commits I-III are basically the signal part.

Commit IV is a hack with a workaround for a tooltip bug :(

Commit V-VI is drag and drop removal, so no one accidentally believes Escoria supports dragging and dropping.

Commit VII fixes an old bug where an item could be permanently stuck in a clicked state (by basically dragging and dropping off it) and therefore prevent the player from moving. Well, technically the player would only move to the same point or another item.

Commit VIII is a tooltip content bug fix.

### CODE REVIEW

File: device/globals/background.gd [hunk 1]
```diff
@@ -4,17 +4,33 @@ export var action = "walk"
...
+			emit_signal("left_click_on_bg", self, pos, event)
 		elif (event.button_index == BUTTON_RIGHT):
-			emit_right_click()
+			emit_signal("right_click_on_bg")
 
 func get_action():
 	return action
 
 func _init():
+	add_user_signal("left_click_on_bg")
```
Comment (StraToN) [inline device/globals/background.gd]: These can simply be defined on top of the script like so:
``` gdscript
signal left_click_on_bg
signal right_click_on_bg
```

File: device/globals/item.gd [hunk 2]
```diff
@@ -479,6 +452,12 @@ func _find_sprites(p = null):
 	for i in range(0, p.get_child_count()):
 		_find_sprites(p.get_child(i))
 
+func _init():
+	add_user_signal("left_click_on_item")
```
Comment (StraToN) [inline device/globals/item.gd]: Same, signals can be defined with `signal` keyword.

File: device/globals/background.gd (hunk 1 above)
Comment (mjtorn) [reply device/globals/background.gd]: Thanks, didn't know that :) Updated.

File: device/globals/item.gd (hunk 2 above)
Comment (mjtorn) [reply device/globals/item.gd]: Thanks, didn't know that :) Updated.

**Pull request merged**